src/javascript/_common/base/socket_base.js

```diff
@@ -289,6 +298,8 @@ const BinarySocketBase = (() => {
         sendBuffered      : sendBufferedRequests,
         get               : () => binary_socket,
         setOnDisconnect   : (onDisconnect) => { config.onDisconnect = onDisconnect; },
+        setOnReconnect    : (onReconnect) => {config.onReconnect = onReconnect; },
+        removeOnReconnect : () => { delete config.onReconnect; },
```

Comment:
why not combine it with removeOnDisconnect? if we are using both in same places

Reply:
We need to do it on `onopen` of the reconnection. onDisconnect only happens when it is disconnected. and in that time, I don't think it'll help if we send the request.